### PR TITLE
Add Pluralization of @NLparameters (#2593)

### PR DESCRIPTION
**Pull request by Pramodh-G**
Fixes Issue #2593 . 
Please suggest ways in which this PR can be improved :)

### CODE REVIEW

**Review by path-doc — COMMENTED** [review]
May I suggest that you make clear the syntax for begin ... end blocks:

"Distinct parameters need to be placed on seperate lines without base_names as in the following example."

This is because other objects (such as variables and constraints) can be assigned names.

**Comment by Pramodh-G** [comment]
Sure, I'll update it.

File: src/macros.jl [hunk 1]
```diff
@@ -1092,6 +1093,30 @@ end)
...
 
+@doc """
+     @NLparameters(model, args...)
+
+ Create and return multiple nonlinear parameters attached to model `model`, in the same fashion as
+ [`@NLparameter`](@ref) macro.
+
+ The model must be the first argument, and multiple parameters can be added on
+ multiple lines wrapped in a `begin ... end` block. Distinct parameters need to be placed
+ on separate lines without `base_names` as in the following example.
```
Comment (blegat) [inline src/macros.jl:1104]: > without `base_names`

This is not clear what you mean by that, I would just remove it

Comment (odow) [inline src/macros.jl:1104]:
```suggestion
 on separate lines as in the following example.
```

**Review by odow — APPROVED** [review]

**Comment by path-doc** [comment]
What was meant is that somewhere it should be clear that, unlike variables
and constraints, parameters cannot be given a ```name::String```.

**Comment by odow** [comment]
> What was meant is that somewhere it should be clear that, unlike variables and constraints, parameters cannot be given a ```name::String```.
> […](#)

This is covered in the docstring of `@NLparameter`: https://jump.dev/JuMP.jl/v0.21.8/reference/nlp/#ref_nl_parameters. Adding extra arguments is a separate issue.

**Comment by path-doc** [comment]
We wouldn't want any redundancy now would we :)

**Comment by odow** [comment]
We link to the `@NLparameter` docstring and explain it follows the same syntax. Moreover, the general rule for docstrings is to explain the things that can be done,  rather than what can't be done.

Error messages should tell what people did wrong (this isn't the greatest example because it actually looks like a bug!:
```Julia
julia> @NLparameter(model, p == 1, base_name = "x")
ERROR: LoadError: At REPL[7]:1: `@NLparameter(model, p == 1)`: Too many arguments.
Stacktrace:
 [1] error(::String, ::String)
   @ Base ./error.jl:42
 [2] _macro_error(macroname::Symbol, args::Tuple{Symbol, Expr}, source::LineNumberNode, str::String)
   @ JuMP ~/.julia/dev/JuMP/src/macros.jl:1365
 [3] (::JuMP.var"#_error#122"{LineNumberNode, Symbol, Expr})(str::String)
   @ JuMP ~/.julia/dev/JuMP/src/macros.jl:2114
 [4] var"@NLparameter"(__source__::LineNumberNode, __module__::Module, m::Any, ex::Any, extra::Vararg{Any, N} where N)
   @ JuMP ~/.julia/dev/JuMP/src/macros.jl:2118
in expression starting at REPL[7]:1
```

**Pull request merged**